### PR TITLE
add policy, policyset and policy config documentations

### LINKED ISSUE #2929: Add PolicySet and PolicyConfig to website docs
After the changes in weaveworks/weave-gitops-enterprise#1540 and
#112 the way we use the policy sets and policies will change a lot. A lot of use cases will be introduced, as-well-as edge cases, adding more complexity to the way we configure the policies and the agent. We want to make it easier for our stakeholders and customers to understand and apply those changes.

Some of the things that should be covered by the docs:

- Using policy sets for each mode and how it affects the policy object.
- An explanation of the policyconfig and how it overrides the policy parameters.
- The validations done on the creation of the policyconfig objects.
-  The different priorities of the policyconfigs and how they override each other.

Those doc changes should be released at the same time with the profile release weaveworks/profiles-catalog#192

### PR DESCRIPTION
**Pull request by waleedhammam**
Closes #2929

**What changed?**

- add policy config documentations

### CODE REVIEW

**Review by MostafaMegahid — APPROVED** [review]

File: website/docs/policy/configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+title: Weave policy configuration
```
Comment (serboctor) [inline website/docs/policy/configuration.mdx:2]:
```suggestion
title: Weave policy agent configuration
```

File: website/docs/policy/releases.mdx [hunk 2]
```diff
@@ -1,8 +1,9 @@
 ---
-title: Profile Releases
-sidebar_position: 4
+title: Weave policy configuration
 hide_title: true
+sidebar_position: 6
```
Comment (serboctor) [inline website/docs/policy/releases.mdx:4]: releases should ideally be the last page

**Review by serboctor — CHANGES_REQUESTED** [review]

**Review by serboctor — APPROVED** [review]
LGTM

File: website/docs/policy/releases.mdx [hunk 3]
```diff
@@ -1,12 +1,39 @@
...
 
 # Profile Releases <TierLabel tiers="enterprise" />
 
+## v0.6.4
+
+### Highlights
+- **Agent**
+  - Add PolicyConfig CRD to make it possible to customize policy configuration per namespaces, applications or resources
+  - Add mode field to policy set and add policy modes to its status
+  - add policy modes to labels to support filtering
```
Comment (serboctor) [inline website/docs/policy/releases.mdx:17]:
```suggestion
  - Add policy modes to labels to support filtering
```

File: website/docs/policy/releases.mdx [hunk 4]
```diff
@@ -1,12 +1,39 @@
...
 # Profile Releases <TierLabel tiers="enterprise" />
 
+## v0.6.4
+
+### Highlights
+- **Agent**
+  - Add PolicyConfig CRD to make it possible to customize policy configuration per namespaces, applications or resources
+  - Add mode field to policy set and add policy modes to its status
+  - Add policy modes to labels to support filtering
+  - support backward compatibility for policy version v2beta1
```
Comment (serboctor) [inline website/docs/policy/releases.mdx:18]:
```suggestion
  - Support backward compatibility for policy version v2beta1
```

**Review by serboctor — APPROVED** [review]

**Pull request merged**